src/Index/AbstractAggregateIndex.php

```diff
@@ -42,6 +42,9 @@ protected function registerIndex(ReadableIndex $index)
         $index->on('definition-added', function () {
             $this->emit('definition-added');
         });
+        $index->on('wipe', function() {
```

Comment:
The runtime code shouldn't emit events just so tests can listen to this - if a method call is to be asserted, the object should be mocked. Atm I do this manually in some tests, a better way would be to use a mocking library (PHPUnit or prophecy)